dev/io.openliberty.security.jakartasec.4.0.internal/src/io/openliberty/security/jakartasec/identitystore/InMemoryIdentityStoreDefinitionWrapper.java

```diff
@@ -0,0 +1,421 @@
+             * again later.
+             */
+            if (immediateOnly && ELHelperWrapper.isDeferredExpression(useForExpression)) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "evaluateUseFor", "Returning null since useForExpression is a deferred expression and this is called on initialization.");
+                }
+                return null;
+            }
+
+            if (TraceComponent.isAnyTracingEnabled() && tc.isWarningEnabled()) {
+                Tr.warning(tc, "JAKARTASEC_WARNING_IDSTORE_CONFIG", new Object[] { "useFor/useForExpression", JakartaSec40Constants.SPEC_DEFAULT_USEFOR });
+            }
+
+            Set<ValidationType> values = new HashSet<ValidationType>();
+            values.addAll(Set.of(JakartaSec40Constants.SPEC_DEFAULT_USEFOR));
+
+            return values;
+        }
+    }
+
+    /**
+     * Get the priority for the {@link IdentityStore}.
+     *
+     * @return The priority.
+     *
+     * @see InMemoryIdentityStoreDefinition#priority()
+     * @see InMemoryIdentityStoreDefinition#priorityExpression()
+     */
+    int getPriority() {
+        return (this.priority != null) ? this.priority : evaluatePriority(false);
+    }
+
+    /**
+     * Get the useFor for the {@link IdentityStore}.
+     *
+     * @return The useFor.
+     *
+     * @see InMemoryIdentityStoreDefinition#useFor()
+     * @see InMemoryIdentityStoreDefinition#useForExpression()
+     */
+    Set<ValidationType> getUseFor() {
+        return (this.useFor != null) ? this.useFor : evaluateUseFor(false);
+    }
+
+    /*-
+     * Get the Credentials for the {@link IdentityStore} as a {@HashMap}.
+     *
+     * @return The CredentialValue as a {@HashMap}, with each Map element
+     *         keyed on the callerName.  An an empty {@HashMap} is returned
+     *         if no credentials available.
+     *
+     *         NOTE: The caller password is stored internally based on
+     *         if it was an encoded or hashed value.  This internal representation
+     *         is not programatically visible to the client, but the validate()
+     *         method can be used for safe password comparisons.
+     *
+     * @see InMemoryIdentityStoreDefinition#CredentialValue()
+     */
+    Map<String, CredentialValue> getCredentials1() {
+        return ((credentials == null) || (credentials.size() != 0)) ? credentials : new HashMap<String, CredentialValue>(0);
+    }
+
+    /**
+     * Get a specific credential from the credentials map given a caller name.
+     *
+     * @param callerName is the caller name key to the map.
+     * @return the CredentialValue, or null if the caller name is invalid.
+     */
+    public CredentialValue getCredential(String callerName) {
+        return ((credentials == null) || (credentials.size() != 0)) ? credentials.get(callerName) : null;
```

Comment:
```suggestion
        return credentials.get(callerName);
```
Here and in other places you are checking for `null`.  It is new'ed up in the constructor so not sure how it can be null.  If it were null the logic in this condition would generate a NullPointerException.

Reply:
Yep, updated as suggested based.

Weird, just weird, the boolean logic I wrote.  Hmmmmm